packages/medusa/src/loaders/models.ts

```diff
@@ -5,30 +5,76 @@ import { ClassConstructor, MedusaContainer } from "../types/global"
 import { EntitySchema } from "typeorm"
 import { asClass, asValue } from "awilix"
 
+function formatModelName(modelName) {
+  // Remove "Model" from the end of the string
+  let formattedName = modelName.replace('Model', '');
```

Comment:
question: Where is model coming from?

Reply:
Maybe it is just in case?

Reply:
Unsure where up the chain it comes from, but the modelName contains it. like `productCategoryModel`

Reply:
It comes from the formatRegistrationName which append the namespace to the original formatted name.

Reply:
Maybe you can update the `format-registration-name` file utils to something like this
```ts
// packages/medusa/src/utils/format-registration-name.ts
import { parse } from "path"

/**
 * Formats a filename into the correct container resolution name.
 * Names are camelCase formatted and namespaced by the folder i.e:
 * models/example-person -> examplePersonModel
 * @param path - the full path of the file
 * @return the formatted name
 */
function formatRegistrationName(path: string): string {
  const parsed = parse(path)
  const parsedDir = parse(parsed.dir)

  let namespace = parsedDir.name
  if (namespace.startsWith("__")) {
    const parsedCoreDir = parse(parsedDir.dir)
    namespace = parsedCoreDir.name
  }

  switch (namespace) {
    // We strip the last character when adding the type of registration
    // this is a trick for plural "ies"
    case "repositories":
      namespace = "repositorys"
      break
    case "strategies":
      namespace = "strategys"
      break
    default:
      break
  }

  const upperNamespace =
    namespace.charAt(0).toUpperCase() + namespace.slice(1, -1)

  return formatRegistrationNameWithoutNamespace(path) + upperNamespace
}

export function formatRegistrationNameWithoutNamespace(path: string): string {
  const parsed = parse(path)

  const rawname = parsed.name

  const parts = rawname.split("-").map((n, index) => {
    if (index !== 0) {
      return n.charAt(0).toUpperCase() + n.slice(1)
    }
    return n
  })

  return parts.join("")
}

export default formatRegistrationName
```

Reply:
And when needed you can re use the `upperCaseFirst` I have created in the module CLI [pr](https://github.com/medusajs/medusa/pull/4378/files#diff-e7c87478ac5906f5d9701e55378173848b7ef8e051ff6b203555209fa82973b3) instead of having the charAt(0) everywhere 😂